src/libraries/System.Private.DataContractSerialization/src/System/Runtime/Serialization/ClassDataContract.cs

```diff
@@ -389,60 +318,15 @@ private static bool IsArraySegment(Type t)
 
             if (type.IsDefined(Globals.TypeOfDataContractAttribute, false))
                 return false;
-            if (type.IsValueType)
-            {
-                return type.IsVisible;
-            }
-            else
-            {
-                return (type.IsVisible &&
-                    type.GetConstructor(BindingFlags.Instance | BindingFlags.NonPublic | BindingFlags.Public, Type.EmptyTypes) != null);
-            }
-        }
-
-        private static readonly Dictionary<string, string[]> s_knownSerializableTypeInfos = new Dictionary<string, string[]> {
-            { "System.Collections.Generic.KeyValuePair`2", Array.Empty<string>() },
```

Comment:
I think this list is here to compensate for the Serializable attribute having been removed. For example, KeyValuePair is attributed in NetFx, but it isn't in .NET. They did go back and re-add back the Serializable attribute to many types. A few types in this list that I checked have it back, the only one I found where they didn't was KeyValuePair, but I only checked a few of them.

Reply:
Yes, all the types have Serializable added back, including [KeyValuePair`2](https://source.dot.net/#System.Private.CoreLib/KeyValuePair.cs,27).